server/internal/core/application/covenantless.go

```diff
@@ -0,0 +1,1004 @@
+	if len(vtxos) > 0 {
+		s.stopWatchingVtxos(vtxos)
+	}
+
+	s.wallet.Close()
+	log.Debug("closed connection to wallet")
+	s.repoManager.Close()
+	log.Debug("closed connection to db")
+	close(s.eventsCh)
+	close(s.onboardingCh)
+}
+
+func (s *covenantlessService) SpendVtxos(ctx context.Context, inputs []domain.VtxoKey) (string, error) {
+	vtxos, err := s.repoManager.Vtxos().GetVtxos(ctx, inputs)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range vtxos {
+		if v.Spent {
+			return "", fmt.Errorf("input %s:%d already spent", v.Txid, v.VOut)
+		}
+	}
+
+	payment, err := domain.NewPayment(vtxos)
+	if err != nil {
+		return "", err
+	}
+	if err := s.paymentRequests.push(*payment); err != nil {
+		return "", err
+	}
+	return payment.Id, nil
+}
+
+func (s *covenantlessService) ClaimVtxos(ctx context.Context, creds string, receivers []domain.Receiver) error {
+	// Check credentials
+	payment, ok := s.paymentRequests.view(creds)
+	if !ok {
+		return fmt.Errorf("invalid credentials")
+	}
+
+	if err := payment.AddReceivers(receivers); err != nil {
+		return err
+	}
+	return s.paymentRequests.update(*payment)
+}
+
+func (s *covenantlessService) UpdatePaymentStatus(_ context.Context, id string) ([]string, *domain.Round, error) {
+	err := s.paymentRequests.updatePingTimestamp(id)
+	if err != nil {
+		if _, ok := err.(errPaymentNotFound); ok {
+			return s.forfeitTxs.view(), s.currentRound, nil
+		}
+
+		return nil, nil, err
+	}
+
+	return nil, nil, nil
+}
+
+func (s *covenantlessService) SignVtxos(ctx context.Context, forfeitTxs []string) error {
+	return s.forfeitTxs.sign(forfeitTxs)
+}
+
+func (s *covenantlessService) ListVtxos(ctx context.Context, pubkey *secp256k1.PublicKey) ([]domain.Vtxo, []domain.Vtxo, error) {
+	pk := hex.EncodeToString(pubkey.SerializeCompressed())
+	return s.repoManager.Vtxos().GetAllVtxos(ctx, pk)
+}
+
+func (s *covenantlessService) GetEventsChannel(ctx context.Context) <-chan domain.RoundEvent {
+	return s.eventsCh
+}
+
+func (s *covenantlessService) GetRoundByTxid(ctx context.Context, poolTxid string) (*domain.Round, error) {
+	return s.repoManager.Rounds().GetRoundWithTxid(ctx, poolTxid)
+}
+
+func (s *covenantlessService) GetRoundById(ctx context.Context, id string) (*domain.Round, error) {
+	return s.repoManager.Rounds().GetRoundWithId(ctx, id)
+}
+
+func (s *covenantlessService) GetCurrentRound(ctx context.Context) (*domain.Round, error) {
+	return domain.NewRoundFromEvents(s.currentRound.Events()), nil
+}
+
+func (s *covenantlessService) GetInfo(ctx context.Context) (*ServiceInfo, error) {
+	pubkey := hex.EncodeToString(s.pubkey.SerializeCompressed())
+
+	return &ServiceInfo{
+		PubKey:              pubkey,
+		RoundLifetime:       s.roundLifetime,
+		UnilateralExitDelay: s.unilateralExitDelay,
+		RoundInterval:       s.roundInterval,
+		Network:             s.network.Name,
+		MinRelayFee:         int64(s.minRelayFee),
+	}, nil
+}
+
+// TODO clArk changes the onboard flow (2 rounds ?)
```

Comment:
if we use the simple "boarding wihtout luggage" its one-step boarding, user sends to address.

```
- A + S 
- A after 3 months 
```

Reply:
hm yes but Alice and ASP need an extra round to cosign together the tree no ?